src/main/java/com/openlattice/datastore/search/controllers/SearchController.java

```diff
@@ -170,11 +168,11 @@ public DataSearchResult searchEntitySetData( @RequestBody SearchConstraints sear
             results = searchService.executeSearch( searchConstraints, authorizedPropertyTypesByEntitySet );
         }
 
-        List<AuditableEvent> searchEvents = new ArrayList( entitySetIds.length );
-        for ( int i = 0; i < entitySetIds.length; i++ ) {
+        List<AuditableEvent> searchEvents = Lists.newArrayListWithExpectedSize( entitySetIds.length );
```

Comment:
Should actually use `new ArrayList( entitySetIds.length );` here since you know exactly how many elements are being added. `newArrayListWithExpectedSize` does some guesstimates for how to size the array that are un-necessary when you know exactly how big the list is going to be 👍